packages/client-config/API.md

```diff
@@ -210,18 +210,11 @@ export type CustomClientConfig = {
 export const DEFAULT_CLIENT_CONFIG_VERSION: ClientConfigVersion;
 
 // @public
-export const generateClientConfig: <T extends "0" | "1">(awsClientProvider: AWSClientProvider<{
+export const generateClientConfig: <T extends "1" | "0">(backendIdentifier: DeployedBackendIdentifier, version: T, awsClientProvider: AWSClientProvider<{

```

Comment:
can you move this param at the end in the second api too ?

Reply:
This one would need to be optional in order to move it to the end. Should I make it optional and instantiate new ones when its not specified?

Reply:
> Should I make it optional and instantiate new ones when its not specified?

Yes let's do this. (and I think we should do the same for the other API to make them look consistent)

Reply:
not trying to cause too much churn here, but I actually prefer the client provider as a required first arg. It keeps it from getting muddled in with the other args that are more related to business logic

Reply:
My reasoning is that from caller POV these clients could be optional and simplify call site if they can be newed up on defaults. (in generic case outside of console need).
And the reason I'd like to simplify basic use case is that this API was assumed customer facing in early specs.